src/test/clojure/clara/test_rules.clj

```diff
@@ -1133,6 +1133,17 @@
                fire-rules
                (query not-different-temps))))))
 
+(deftest test-empty-test-condition
+
+  (is (thrown-with-msg?
+        clojure.lang.ExceptionInfo
+        #"line.*123.*column.*456"
```

Comment:
Is there a particular reason why you didn't use [assert-ex-data](https://github.com/cerner/clara-rules/blob/0.16.0/src/test/clojure/clara/test_rules.clj#L76) here or were you just not aware of it?  It has the advantages of:

- Using normal Clojure data equality semantics rather than messy string comparisons.
- Not failing if something else wraps the exception later since it walks the exception chain to look for a match.

Side note: Clojure has an [IExceptionInfo interface](https://github.com/clojure/clojure/blob/clojure-1.8.0/src/jvm/clojure/lang/IExceptionInfo.java), of which [clojure.lang.ExceptionInfo](https://github.com/clojure/clojure/blob/clojure-1.8.0/src/jvm/clojure/lang/ExceptionInfo.java) is a concrete implementation.  Any instance checks you do should generally use IExceptionInfo for all the usual reasons to favor coding against an interface rather than the implementation; in this particular case my understanding is that user programs might define implementations of IExceptionInfo themselves.  The [ex-data function](http://clojuredocs.org/clojure.core/ex-data) just returns nil if the exception is of the wrong type so that can be used as well in many cases in logic like

````
(when (ex-data e) ...
````

Reply:
I wasnt aware of it. I copied a test elsewhere in this file and modified it for my needs. I'll work it out to use this.